sdk/src/client/api/block_builder/input_selection/core/requirement/amount.rs

```diff
@@ -308,7 +334,25 @@ impl InputSelection {
             return Ok(r);
         }
 
-        if self.selected_inputs.len() + amount_selection.newly_selected_inputs.len() > INPUT_COUNT_MAX.into() {
+        // If the available inputs have more NTs than are allowed in a single tx, we might not be able to find inputs
+        // without exceeding the threshold, so in this case we also try again with the outputs ordered the other way
+        // around.
+        let potentially_too_many_native_tokens = HashSet::<TokenId>::from_iter(
+            self.available_inputs
```

Comment:
Why available ?

Reply:
Because we don't know yet which inputs we need

Reply:
If the available inputs already have less than max NTs, then doing a second round will not change anything, since this can't be the problem